ando/tools/generator/nwb2bidsgenerator.py

```diff
@@ -0,0 +1,214 @@
+        else will create a symlink
+    output_path: [str, Path]
+        parent path to the BIDS folder structure
+    dataset_path : [str, Path]
+        path to the folder containing all the nwb datasets that need organization.
+    validate: bool
+        to validate the dataset using the ANdoChecker
+    kwargs:
+        probe_type: acute/chronic
+    """
+    dataset_path = Path(dataset_path)
+    if output_path is None:
+        output_path = dataset_path.parent/'BIDSExt'/dataset_path.name
+    if re_write and output_path.exists():
+        output_path.unlink()
+    participants_df = pd.DataFrame(
+        columns=['Species', 'ParticipantID', 'Sex', 'Birthdate', 'Age', 'Genotype', 'Weight'])
+    dataset_desc_json = None
+
+    dataset_path = Path(dataset_path)
+    file_count = 0
+    sessions_count = 0
+    sub_ses_dict = defaultdict(list)
+    for nwb_file in dataset_path.glob('**/*.nwb'):
+        file_count += 1
+        channels_df = pd.DataFrame(
+            columns=['channel_id', 'Contact_id', 'type', 'units', 'sampling_frequency',
+                     'unit_conversion_multiplier'])
+        contacts_df = pd.DataFrame(
+            columns=[
+                'x',
+                'y',
+                'z',
+                'impedance',
+                'contact_id',
+                'probe_id',
+                'Location'])
+        probes_df = pd.DataFrame(columns=['probeID', 'type'])
+
+        with NWBHDF5IO(str(nwb_file), 'r') as io:
+            nwbfile = io.read()
+
+            # subject info:
+            if nwbfile.subject is not None:
+                sb = nwbfile.subject
+                if sb.subject_id is not None:
+                    subject_label = f'sub-{sb.subject_id}'
+                else:
+                    subject_label = f'sub-{sb.date_of_birth.strftime("%Y%m%dT%H%M")}'
+                if not participants_df['ParticipantID'].str.contains(
+                        subject_label).any():
+                    participants_df.loc[len(participants_df.index)] = \
+                        [sb.species, subject_label, sb.sex[0] if sb.sex is not None else None,
+                         sb.date_of_birth, sb.age, sb.genotype, sb.weight]
+            else:
+                subject_label = 'sub-none'
+                if not participants_df['ParticipantID'].str.contains(
+                        subject_label).any():
+                    participants_df.loc[len(participants_df.index)] = \
+                        [None, subject_label, None, None, None, None, None]
+
+            # dataset info:
+            if dataset_desc_json is None:
+                dataset_desc_json = dict(
+                    InstitutionName=nwbfile.institution, InstitutionalDepartmentName=nwbfile.lab,
+                    Name='Electrophysiology', BIDSVersion='1.0.X',
+                    Licence='CC BY 4.0',
+                    Authors=[
+                        list(nwbfile.experimenter) if nwbfile.experimenter is not None else None])
```

Comment:
This is going to be a nested list, but should rather be a flat one, no?

Reply:
fixed